harness/determined/deploy/gke/cli.py

```diff
@@ -0,0 +1,551 @@
+        "saveExperimentBest", 0
+    )
+    checkpointStorage["saveTrialBest"] = helm_values["checkpointStorage"].get("saveTrialBest", 1)
+    checkpointStorage["saveTrialLatest"] = helm_values["checkpointStorage"].get(
+        "saveTrialLatest", 1
+    )
+    checkpointStorage["type"] = "gcs"
+    checkpointStorage["bucket"] = args.gcs_bucket
+    helm_values["checkpointStorage"] = checkpointStorage
+    helm_values["maxSlotsPerPod"] = args.gpus_per_node
+    if args.multi_node_pool:
+        gpu_pod_spec = make_spec(helm_values["taskContainerDefaults"], "gpuPodSpec")
+        gpu_pod_spec["spec"]["tolerations"] = [
+            {
+                "key": "gpuAvailable",
+                "operator": "Equal",
+                "value": "True",
+                "effect": "NoSchedule",
+            }
+        ]
+        gpu_pod_spec["spec"]["nodeSelector"] = {"accelerator_type": "gpu"}
+        helm_values["taskContainerDefaults"]["gpuPodSpec"] = gpu_pod_spec
+        cpu_pod_spec = make_spec(helm_values["taskContainerDefaults"], "cpuPodSpec")
+        cpu_pod_spec["spec"]["tolerations"] = [
+            {
+                "key": "gpuAvailable",
+                "operator": "Equal",
+                "value": "False",
+                "effect": "NoSchedule",
+            }
+        ]
+        cpu_pod_spec["spec"]["nodeSelector"] = {"accelerator_type": "cpu"}
+        if args.cpu_coscheduler:
+            cpu_pod_spec["spec"]["schedulerName"] = "coscheduler"
+        helm_values["taskContainerDefaults"]["cpuPodSpec"] = cpu_pod_spec
+    with (helm_dir / "values.yaml").open("w") as f:
+        yaml.round_trip_dump(helm_values, f)
+
+
+def handle_up(args: argparse.Namespace) -> None:
+    try:
+        validate_args(args)
+    except Exception as e:
+        exc_str = "Argument Error: {}".format(e)
+        cprint(exc_str, "red")
+        cprint("Failed to create gke cluster", "red")
+        sys.exit(1)
+    if args.gpu_coscheduler or args.preemption:
+        cprint(
+            (
+                "Autoscaling is not supported with the lightweight coscheduling plugin or with the"
+                + " preemptive priority-based scheduler, and so a GPU node pool with {} nodes will"
+                + " be statically allocated. This can be changed by using the default Kubernetes"
+                + "  scheduler, or specifying a different value for --max-gpu-nodes."
+            ).format(args.max_gpu_nodes),
+            "yellow",
+        )
+    if args.cpu_coscheduler:
+        cprint(
+            (
+                "Autoscaling is not supported with the lightweight coscheduling plugin and so a CPU"
+                + " node pool with {} nodes will be statically allocated. This can be changed by"
+                + "  using the default Kubernetes scheduler, or specifying a different value for"
+                + " --max-cpu-nodes."
+            ).format(args.max_cpu_nodes),
+            "yellow",
+        )
+    create_cluster(args)
```

Comment:
what happens when you try to "upgrade" existing cluster?

Reply:
I'm not sure I follow the question - do you mean update the settings or the helm chart?

Reply:
either: a user runs `det deploy gke` with the same cluster id and the same or different flags. won't gcloud for cluster create fail in that case?

Reply:
Ah yeah - that would be an issue but only if they're using the same project. I'm not sure it makes sense for the same user to be able to create multiple gke clusters with the same name - or did you mean for that to be a shortcut to tear down the existing cluster, and create a new one with the different flags?

Reply:
>did you mean for that to be a shortcut to tear down the existing cluster, and create a new one with the different flags?
yes. or we can just show an error "cannot do that, please delete the cluster first".

when only helm chart parameters are updated, and we're upgrading the existing cluster, then we can skip the gke cluster portion, and just do `helm upgrade`.

Reply:
following up: what happens if you run the same creation command twice? does that error look reasonable?